### PR TITLE
Proposals/total roof area

### PR DESCRIPTION
**Pull request by JieXiong9119**
See proposal
Implementation in schema added

### CODE REVIEW

File: BuildingSync.xsd [hunk 1]
```diff
@@ -796,6 +796,18 @@
           </xs:simpleContent>
         </xs:complexType>
       </xs:element>
+      <xs:element name="TotalExteriorRoofArea" minOccurs="0">
```
Comment (macintoshpie) [inline BuildingSync.xsd:799]: Wondering if we should use `TotalRoofArea` instead? I don't have a preference, was wondering if there might have been a reason we added `Exterior` to the wall area one, but if it is necessary for roofs.

Comment (JieXiong9119) [reply BuildingSync.xsd:799]: I think total roof area is concise and better. Is there a non-exterior/interior roof anywhere?

Comment (corymosiman12) [reply BuildingSync.xsd:799]: I think `TotalRoofArea` is good.

File: BuildingSync.xsd [hunk 2]
```diff
@@ -796,6 +796,18 @@
...
+        <xs:annotation>
+          <xs:documentation>Total roof area exposed to the elements. (ft2)</xs:documentation>
+        </xs:annotation>
+        <xs:complexType>
+          <xs:simpleContent>
+            <xs:extension base="xs:decimal">
```
Comment (corymosiman12) [inline BuildingSync.xsd:805]: maybe make this as a non negative decimal type (we might need to add this as a base type). see the `BoundedDecimalZeroToOne` for something similar.

**Review by corymosiman12 — CHANGES_REQUESTED** [review]
- create new simple type for nonNegativeDecimal, something like:
```xml
<xs:simpleType name="BoundedDecimalZeroToOne">
    <xs:restriction base="xs:decimal">
      <xs:minInclusive value="0"/>
      <xs:maxInclusive value="1"/>
    </xs:restriction>
  </xs:simpleType>
```
- Use `TotalRoofArea`

Else looks good!

**Comment by JieXiong9119** [comment]
> * create new simple type for nonNegativeDecimal, something like:
> 
> ```
> <xs:simpleType name="BoundedDecimalZeroToOne">
>     <xs:restriction base="xs:decimal">
>       <xs:minInclusive value="0"/>
>       <xs:maxInclusive value="1"/>
>     </xs:restriction>
>   </xs:simpleType>
> ```
> 
> * Use `TotalRoofArea`
> 
> Else looks good!

Done! Please check it again.
Should we rename the PR as well?

**Comment by corymosiman12** [comment]
Perfect. Yeah, I think renaming would be good.

**Review by yan130 — APPROVED** [review]

**Review by corymosiman12 — APPROVED** [review]
👍

**Comment by nllong** [comment]
@JieXiong9119 -- can you resolve the conflicts on this one?

**Pull request merged**